`PythonRunIntegrationTest.test_run_27_and_then_26` failing in master

seen e.g. [here](https://travis-ci.org/pantsbuild/pants/jobs/274521259#L2871):

```
                     ==================== FAILURES ====================
                      PythonRunIntegrationTest.test_run_27_and_then_26 
                     
                     self = <pants_test.backend.python.tasks2.test_python_run_integration.PythonRunIntegrationTest testMethod=test_run_27_and_then_26>
                     
                         def test_run_27_and_then_26(self):
                           with temporary_dir() as interpreters_cache:
                             pants_ini_config = {'python-setup': {'interpreter_cache_dir': interpreters_cache}}
                             pants_run_27 = self.run_pants(
                               command=['run', '{}:echo_interpreter_version_2.7'.format(self.testproject)],
                               config=pants_ini_config
                             )
                             self.assert_success(pants_run_27)
                             pants_run_26 = self.run_pants(
                               command=['run', '{}:echo_interpreter_version_2.6'.format(self.testproject),
                                        '--python-setup-interpreter-constraints=CPython>=2.6,<3',
                                        '--python-setup-interpreter-constraints=CPython>=3.3'],
                               config=pants_ini_config
                             )
                     >       self.assert_success(pants_run_26)
                     
                     .pants.d/pyprep/sources/344a98c9d8c37784a8933e6d733d3ae7018fb0ea/pants_test/backend/python/tasks2/test_python_run_integration.py:35: 
                     _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
                     .pants.d/pyprep/sources/344a98c9d8c37784a8933e6d733d3ae7018fb0ea/pants_test/pants_run_integration_test.py:315: in assert_success
                         self.assert_result(pants_run, self.PANTS_SUCCESS_CODE, expected=True, msg=msg)
                     .pants.d/pyprep/sources/344a98c9d8c37784a8933e6d733d3ae7018fb0ea/pants_test/pants_run_integration_test.py:338: in assert_result
                         assertion(value, pants_run.returncode, error_msg)
                     E   AssertionError: /home/travis/build/pantsbuild/pants/pants --no-pantsrc --pants-workdir=/home/travis/build/pantsbuild/pants/.pants.d/tmp/tmpEeo_SW.pants.d --kill-nailguns --print-exception-stacktrace --pants-config-files=/home/travis/build/pantsbuild/pants/.pants.d/tmp/tmpEeo_SW.pants.d/pants.ini run testprojects/src/python/interpreter_selection:echo_interpreter_version_2.6 --python-setup-interpreter-constraints=CPython>=2.6,<3 --python-setup-interpreter-constraints=CPython>=3.3
                     E   returncode: 1
                     E   stdout:
                     E   	
                     E   	22:19:31 00:00 [main]
                     E   	               (To run a reporting server: ./pants server)
                     E   	22:19:31 00:00   [setup]
                     E   	22:19:32 00:01     [parse]
                     E   	               Executing tasks in goals: jvm-platform-validate -> bootstrap -> imports -> unpack-jars -> deferred-sources -> gen -> resolve -> resources -> compile -> pyprep -> binary -> run
                     E   	22:19:32 00:01   [jvm-platform-validate]
                     E   	22:19:32 00:01     [jvm-platform-validate]
                     E   	22:19:32 00:01   [bootstrap]
                     E   	22:19:32 00:01     [substitute-aliased-targets]
                     E   	22:19:32 00:01     [jar-dependency-management]
                     E   	22:19:32 00:01     [bootstrap-jvm-tools]
                     E   	22:19:32 00:01     [provide-tools-jar]
                     E   	22:19:32 00:01   [imports]
                     E   	22:19:32 00:01     [ivy-imports]
                     E   	22:19:32 00:01   [unpack-jars]
                     E   	22:19:32 00:01     [unpack-jars]
                     E   	22:19:32 00:01     [unpack-libs]
                     E   	22:19:32 00:01   [deferred-sources]
                     E   	22:19:32 00:01     [deferred-sources]
                     E   	22:19:32 00:01   [gen]
                     E   	22:19:32 00:01     [antlr-java]
                     E   	22:19:32 00:01     [antlr-py]
                     E   	22:19:32 00:01     [jaxb]
                     E   	22:19:32 00:01     [protoc]
                     E   	22:19:32 00:01     [ragel]
                     E   	22:19:32 00:01     [thrift-java]
                     E   	22:19:32 00:01     [thrift-py]
                     E   	22:19:32 00:01     [wire]
                     E   	22:19:32 00:01     [aapt]
                     E   	22:19:32 00:01     [go-thrift]
                     E   	22:19:32 00:01     [jax-ws]
                     E   	22:19:32 00:01     [scrooge]
                     E   	22:19:32 00:01   [resolve]
                     E   	22:19:32 00:01     [ivy]
                     E   	22:19:33 00:02     [go]
                     E   	22:19:33 00:02     [scala-js-compile]
                     E   	22:19:33 00:02     [scala-js-link]
                     E   	22:19:33 00:02     [node]
                     E   	22:19:33 00:02   [resources]
                     E   	22:19:33 00:02     [prepare]
                     E   	22:19:33 00:02     [services]
                     E   	22:19:33 00:02   [compile]
                     E   	22:19:33 00:02     [node]
                     E   	22:19:33 00:02     [compile-jvm-prep-command]
                     E   	22:19:33 00:02       [jvm_prep_command]
                     E   	22:19:33 00:02     [compile-prep-command]
                     E   	22:19:33 00:02     [compile]
                     E   	22:19:33 00:02     [zinc]
                     E   	22:19:33 00:02     [jvm-dep-check]
                     E   	22:19:33 00:02     [cpp]
                     E   	22:19:33 00:02     [errorprone]
                     E   	22:19:33 00:02     [findbugs]
                     E   	22:19:33 00:02     [go]
                     E   	22:19:33 00:02   [pyprep]
                     E   	22:19:33 00:02     [interpreter]
                     E   	                   Invalidated 2 targets.
                     E   	FAILURE: Unable to detect a suitable interpreter for compatibilities: CPython>=2.6,<2.7 && CPython>=2.6,<3 (Conflicting targets: testprojects/src/python/interpreter_selection:echo_interpreter_version_2.6, testprojects/src/python/interpreter_selection:echo_interpreter_version_lib)
                     E   	
                     E   	
                     E   	22:19:39 00:08   [complete]
                     E   	               FAILURE
                     ====== 1 failed, 1 passed in 26.08 seconds =======
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -54,6 +54,7 @@ matrix:
 
     - os: linux
       dist: trusty
+      group: deprecated-2017Q2
       sudo: required
       # Docker runs will write files as root, so avoid caching for this shard.
       cache: false
@@ -85,6 +86,7 @@ matrix:
 
     - os: linux
       dist: trusty
+      group: deprecated-2017Q2
       sudo: required
       addons:
         apt:
@@ -113,6 +115,7 @@ matrix:
 
     - os: linux
       dist: trusty
+      group: deprecated-2017Q2
       sudo: required
       addons:
         apt:
@@ -141,6 +144,7 @@ matrix:
 
     - os: linux
       dist: trusty
+      group: deprecated-2017Q2
       sudo: required
       addons:
         apt:
@@ -169,6 +173,7 @@ matrix:
 
     - os: linux
       dist: trusty
+      group: deprecated-2017Q2
       sudo: required
       addons:
         apt:
@@ -197,6 +202,7 @@ matrix:
 
     - os: linux
       dist: trusty
+      group: deprecated-2017Q2
       sudo: required
       addons:
         apt:
@@ -225,6 +231,7 @@ matrix:
 
     - os: linux
       dist: trusty
+      group: deprecated-2017Q2
       sudo: required
       addons:
         apt:
@@ -253,6 +260,7 @@ matrix:
 
     - os: linux
       dist: trusty
+      group: deprecated-2017Q2
       sudo: required
       addons:
         apt:
@@ -281,6 +289,7 @@ matrix:
 
     - os: linux
       dist: trusty
+      group: deprecated-2017Q2
       sudo: required
       addons:
         apt:
@@ -309,6 +318,7 @@ matrix:
 
     - os: linux
       dist: trusty
+      group: deprecated-2017Q2
       sudo: required
       addons:
         apt:
@@ -337,6 +347,7 @@ matrix:
 
     - os: linux
       dist: trusty
+      group: deprecated-2017Q2
       sudo: required
       addons:
         apt:
@@ -365,6 +376,7 @@ matrix:
 
     - os: linux
       dist: trusty
+      group: deprecated-2017Q2
       sudo: required
       addons:
         apt:
@@ -393,6 +405,7 @@ matrix:
 
     - os: linux
       dist: trusty
+      group: deprecated-2017Q2
       sudo: required
       addons:
         apt:

